Mathlib/GroupTheory/Complement.lean

```diff
@@ -236,6 +271,14 @@ theorem card_right_transversal (h : S ∈ rightTransversals (H : Set G)) : Nat.c
     (Equiv.ofBijective _ <| mem_rightTransversals_iff_bijective.mp h).trans <|
       QuotientGroup.quotientRightRelEquivQuotientLeftRel H
 
+@[to_additive]
+lemma finite_of_mem_leftTransversals [H.FiniteIndex] (hs : S ∈ leftTransversals H) : S.Finite :=
+  Nat.finite_of_card_ne_zero <| by rw [card_left_transversal hs]; exact FiniteIndex.finiteIndex
+
+@[to_additive]
+lemma finite_of_mem_rightTransversals [H.FiniteIndex] (hs : S ∈ rightTransversals H) : S.Finite :=
+  Nat.finite_of_card_ne_zero <| by rw [card_right_transversal hs]; exact FiniteIndex.finiteIndex
+
```

Comment:
Don't we already have `Subgroup.MemLeftTransversals.finite_iff`?

Reply:
Honestly I find the existing API hard to read:
1. Why do we have `Subgroup.IsComplement'` instead of using the general `IsCompl` predicate?
2. Why do we have all three of `IsComplement s t`, `s \in leftTransversals t`, `t \in rightTransversals s`? I see that later in the file we talk about `leftTransversals t`/`rightTransversals s` as types, but then why aren't they defined as types in the first place?

Reply:
Yeah, I agree that the API is not so good. I can answer (1) though. `IsCompl` is not quite the same as `Subgroup.IsComplement'`. `IsCompl H K` means that `H` and `K` have trivial intersection and generate `G`. Whereas `IsComplement' H K` means that `H` and `K` have trivial intersection and satisfy `H * K = G`. The latter is what's needed for semidirect products. `IsCompl'` can be satisfied by very small subgroups that happen to generate a larger group `G`.

Reply:
Okay I was suspecting as much. But then shouldn't `Subgroup.IsComplement'` be named something else, since it's stronger than being complement subgroups.

Reply:
In group theory, complement is used for this stronger `HK=G` propery: https://en.wikipedia.org/wiki/Complement_(group_theory)

Reply:
🙈